internal/gcp/secret-manager.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2020 - MinIO, Inc. All rights reserved.
+// Use of this source code is governed by the AGPLv3
+// license that can be found in the LICENSE file.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"github.com/minio/kes"
+	"google.golang.org/api/option"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+)
+
+type SecretManager struct {
+	// The GCP project id
```

Comment:
```suggestion
	// The project ID is a unique, user-assigned ID that can be used by Google APIs.
	// The project ID must be a unique string of 6 to 30 lowercase letters, digits, or hyphens.
	// It must start with a letter, and cannot have a trailing hyphen.
	// See: https://cloud.google.com/resource-manager/docs/creating-managing-projects
```

This has to be more elaborate